code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -262,9 +263,22 @@
 			return 0
 
 	if (W.w_class > max_w_class)
-		if(!stop_messages)
-			to_chat(usr, "<span class='notice'>\The [W] is too big for \the [src].</span>")
-		return 0
+		var/yeh = 0
+		if(ignore_w_class.len)
+			for(var/A in ignore_w_class)
+				if(dd_hasprefix(A,"="))
+					// Force strict matching of type.
+					// No subtypes allowed.
+					if("[W.type]"==copytext(A,2))
+						yeh = 1
+						break
+				else if(istype(W, text2path(A) ))
```

Comment:
It seems to be that the ignore_w_class variable has a list of types in string form and he's comparing them in that way. I'm not sure why they're not types in the first place though.